addons/survey/controllers/user_input_session.py

```diff
@@ -0,0 +1,122 @@
+
+    @http.route('/survey/session_next_question/<string:survey_token>', type='json', auth='user', website=True)
+    def survey_session_next_question(self, survey_token, **kwargs):
+        """ This route is called when the host goes to the next question of the session.
+
+        It's not a regular 'request.render' route because we handle the transition between
+        questions using a AJAX call to be able to display a bioutiful fade in/out effect. """
+
+        survey = self._fetch_from_token(survey_token)
+
+        if survey.session_state == 'not_started':
+            # no open session
+            return werkzeug.utils.redirect('/')
+
+        if survey.session_state == 'ready':
+            survey.write({'session_state': 'in_progress'})
+            survey.flush(['session_state'])
+
+        survey.next_question()
+        template_values = self._prepare_manage_session_values(survey)
+        template_values['is_transitioned'] = True
+        return request.env.ref('survey.user_input_session_manage_content').render(template_values).decode('UTF-8')
+
+    @http.route('/survey/session_results/<string:survey_token>', type='json', auth='user', website=True)
+    def survey_session_results(self, survey_token, **kwargs):
+        """ This route is called when the host shows the current question's results.
+
+        It's not a regular 'request.render' route because we handle the display of results using
+        an AJAX request to be able to include the results in the currently displayed page. """
+
+        survey = self._fetch_from_token(survey_token)
+
+        if not survey.session_state:
+            # no open session
+            return werkzeug.utils.redirect('/')
+
+        user_input_lines = request.env['survey.user_input.line'].search([
+            ('user_input_id', 'in', survey.user_input_ids.ids),
+            ('question_id', '=', survey.session_current_question_id.id)
+        ])
+        question_statistics = survey.session_current_question_id._prepare_statistics(user_input_lines)[0]
+
+        return request.env.ref('survey.survey_page_statistics_question').render({
+            'page_record_limit': 10,
+            'hide_question_title': True,
+            'no_print_url': True,
+            'survey': survey,
+            'question': survey.session_current_question_id,
+            'question_data': question_statistics,
+        }).decode('UTF-8')
+
+    @http.route('/survey/session_ranking/<string:survey_token>', type='json', auth='user', website=True)
+    def survey_session_ranking(self, survey_token, **kwargs):
+        """ This route is called when the host shows the current question's attendees ranking.
+
+        It's not a regular 'request.render' route because we handle the display of the ranking
+        using an AJAX request to be able to include the results in the currently displayed page. """
+
+        survey = self._fetch_from_token(survey_token)
+
+        if not survey.session_state:
```

Comment:
? This can't be False as session_state is required.
`if session_state == 'not_started'` or `if not survey` I guess
also, return {}

Reply:
Yep, good catch.
(Actually I'm returning an empty string as this usually returns HTML)